src/upgrade/cve.ts

```diff
@@ -0,0 +1,154 @@
+            .map((cve) => ({
+                id: cve.cve_id || cve.ghsa_id,
+                ghsa_id: cve.ghsa_id,
+                severity: cve.severity,
+                summary: cve.summary,
+                description: cve.description || cve.summary,
+                html_url: cve.html_url,
+                affectedDeps: (cve.vulnerabilities ?? []).map((v) => ({
+                    name: v.package?.name,
+                    vulVersions: v.vulnerable_version_range,
+                    patchedVersion: v.first_patched_version
+                }))
+            }));
+
+        // group the cves by coordinate
+        const depsCves: { dep: string; cves: CVE[]; minVersion?: string | null }[] = [];
+        for (const dep of deps) {
+            const depCves: CVE[] = allCves.filter((cve) => cve.affectedDeps.some((d) => d.name === dep.name));
+            if (depCves.length < 1) {
+                continue;
+            }
+            // find the min patched version for each coordinate
+            let maxPatchedVersion: string | undefined | null;
+            for (const cve of depCves) {
+                const patchedVersion = cve.affectedDeps.find((d) => d.name === dep.name && d.patchedVersion)?.patchedVersion;
+                const coercedPatchedVersion = semver.coerce(patchedVersion);
+                const coercedMaxPatchedVersion = semver.coerce(maxPatchedVersion);
+                if (
+                    !maxPatchedVersion ||
+                    (coercedPatchedVersion &&
+                        coercedMaxPatchedVersion &&
+                        semver.gt(coercedPatchedVersion, coercedMaxPatchedVersion))
+                ) {
+                    maxPatchedVersion = patchedVersion;
+                }
+            }
+
+            depsCves.push({
+                dep: dep.name,
+                cves: depCves,
+                minVersion: maxPatchedVersion
+            });
+        }
+
+        const upgradeIssues =  depsCves.map(depCve => {
+            const currentDep = deps.find(d => d.name === depCve.dep);
+            const mostCriticalCve = findMostCriticalCve(depCve.cves);
+            return {
+                packageId: depCve.dep,
+                packageDisplayName: depCve.dep,
+                currentVersion: currentDep?.version || 'unknown',
+                name: `${mostCriticalCve.id || 'CVE'}`,
+                reason: UpgradeReason.CVE as const,
+                suggestedVersion: {
+                    name: depCve.minVersion || 'unknown',
+                    description: mostCriticalCve.description || mostCriticalCve.summary || 'Security vulnerability detected'
+                },
+                severity: mostCriticalCve.severity,
+                description: mostCriticalCve.description || mostCriticalCve.summary || 'Security vulnerability detected',
+                link: mostCriticalCve.html_url,
+            };
+        });
```

Comment:
please split the logic of getting cves and converting them into UpgradeIssue-s into 2 methods.